### PR TITLE
Update README with examples

### PR DESCRIPTION
**Pull request by bbayles**
Re: Issue #130, this PR updates the README here on GitHub to provide some installation instructions and motivating examples.

We can probably bikeshed this forever, but I think it's an improvement? I'll do the top-level Read The Docs page separately.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
+    >>> list(flatten(iterable))
+    [0, 1, 2, 3]
+
+Several new recipes are available as well:
+
+.. code-block:: python
+
+    >>> from more_itertools import spy, chunked
+    >>> iterable = [0, 1, 2, 3, 4, 5, 6, 7]
+    >>> head, iterable = spy(iterable, n=2)
```
Comment (pylang) [inline README.rst]: I'd show `chunked` first, then `spy`.  Somehow it is very clear what is going on with the `chunked` example without thinking.  This seems like an easier flow from `flatten`.

File: README.rst [hunk 2]
```diff
@@ -1,23 +1,53 @@
-==============
 More Itertools
```
Comment (pylang) [inline README.rst]: I don't think this is needed.

File: README.rst [hunk 3]
```diff
@@ -1,23 +1,53 @@
...
+Getting started
+===============
+
+To get started, install the library with `pip <https://pip.pypa.io/en/stable/>`_:
+
+.. code-block:: shell
+
+    pip install more-itertools
+
+The recipes from the `itertools docs <https://docs.python.org/3/library/itertools.html>`_
```
Comment (pylang) [inline README.rst]: You may instead want the permalink to the recipes section https://docs.python.org/3/library/itertools.html#itertools-recipes

File: README.rst [hunk 4]
```diff
@@ -1,23 +1,53 @@
...
+Python iterables.
+
+Getting started
+===============
+
+To get started, install the library with `pip <https://pip.pypa.io/en/stable/>`_:
+
+.. code-block:: shell
+
+    pip install more-itertools
```
Comment (pylang) [inline README.rst]: If I recall, either dash or underscore (`more-itertools` or `more_itertools`) should work from pip, but only the latter works for imports in python.  For consistency, would it be better to use the latter throughout?

Comment (bbayles) [inline README.rst]: Eh. I think you'll need some more help than we can provide if you're attempting to import something with a `-` in the name.

File: README.rst [hunk 5]
```diff
@@ -5,19 +5,55 @@ More Itertools
...
+
+For the full listing of functions, see the API documentation at `more-itertools.readthedocs.io <https://more-itertools.readthedocs.io/en/latest/api.html>`_.
+
+Development
+===========
+
+``more-itertools`` is maintained by `@erikrose <https://github.com/erikrose>`_
+and `@bbayles <https://github.com/bbayles>`_, with help from `many others <https://github.com/erikrose/more-itertools/graphs/contributors>`_.
+If you have a problem or suggestion, please file a bug or pull request in this
+repository - thanks for contributing!
```
Comment (erikrose) [inline README.rst]: " - thanks" → ". Thanks"

**Comment by erikrose** [comment]
Looks like an improvement to me! It'd be lovely if we could rig a way to not have to copy and paste this onto the front page of the Sphinx docs. Perhaps we can use the [Sphinx include directive](http://reinout.vanrees.org/weblog/2010/12/08/include-external-in-sphinx.html).

**Comment by bbayles** [comment]
Cool - I'll work on something for the docs index also.

**Comment by bbayles** [comment]
Bingo, that worked. Thanks!

**Pull request merged**